securedrop/journalist.py

```diff
@@ -242,9 +242,19 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+        except TypeError as e:
+            if "Non-hexadecimal digit found" in str(e):
+                flash("Invalid secret format: non-hexadecimal digit found")
+            elif "Odd-length string" not in str(e)):
+                flash("Invalid secret format: odd-length string")
+            else:
+                raise
```

Comment:
I think that flashing an "unexpected error" message and also logging the error is preferable to just letting Apache serve a 500 error. Logging is in general way less of a concern on the JI vs the SI, except in the context of certain interactions such as sending a reply to a Source.

Right now, we won't get much out of the fact we're logging these errors because we haven't implemented the opt-in, "securely send your logs to FPF to help us improve SD" feature. However, if/when this changes, logging such unexpected errors will be a means to help us improve exception handling, to catch bugs, and to generally increase code quality.

"Unexpected error" messaging is certainly not great, but maybe slightly better than a 500.